### PR TITLE
fix log_input with tags

### PR DESCRIPTION
**Pull request by annzhang-db**
## What changes are proposed in this pull request?

- Fix how tags are added (extend instead of append) when constructing DatasetInput to preserve correct type
- Modify unit tests to call log_input with tags

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by prithvikannan — APPROVED** [review]
LGTM, thanks @annzhang-db!

File: tests/tracking/fluent/test_fluent.py [hunk 1]
```diff
@@ -1257,7 +1257,7 @@ def test_log_input(tmp_path):
     df.to_csv(path)
     dataset = from_pandas(df, source=path)
     with start_run() as run:
-        mlflow.log_input(dataset, "train")
+        mlflow.log_input(dataset, "train", {"foo": "baz"})
```
Comment (dbczumar) [inline tests/tracking/fluent/test_fluent.py:1260]: Can we also make another call to `log_input` without the tags to ensure coverage of both cases?

File: tests/tracking/fluent/test_fluent.py [hunk 2]
```diff
@@ -1257,7 +1257,7 @@ def test_log_input(tmp_path):
     df.to_csv(path)
```
Comment (dbczumar) [inline tests/tracking/fluent/test_fluent.py:1257]: Can we add a test case for this to `test_rest_tracking.py`? REST tracking was where the issue was first discovered, so it'd be awesome to have some automated CI ensuring that it works :)

Comment (dbczumar) [reply tests/tracking/fluent/test_fluent.py:1257]: We should add a test case that calls `log_input` :D. Given an `MlflowClient` instance, you can access the tracking URI via `client.tracking_uri`. You can then do `mlflow.set_tracking_uri(client.tracking_uri)` and proceed to call `mlflow.log_input()`

Comment (annzhang-db) [reply tests/tracking/fluent/test_fluent.py:1257]: Thank you!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @annzhang-db !

**Pull request merged**